### PR TITLE
TSDB: Expand _id on version conflict

### PR DESCRIPTION
**Pull request by nik9000**
This adds the `_tsid` and `@timestamp` to the version conflict error
message in tsdb mode. I'm trying to debug a version confict problem with
tsdb and think it'd be useful to get this information rather than the
generated `_id`.

### CODE REVIEW

**Comment by nik9000** [comment]
I'd love to get folk's opinion on this one, especially @henningandersen. The new error message looks like:
```
        "reason" : "[IEgWvIN2ZvUEKfcZAJzvEn4BAAA/{uid=foo}@2022-01-01T00:00:00.000Z]: version conflict, document already exists (current version [1])",
```

**Comment by nik9000** [comment]
@henningandersen I think this one is ready for you.

**Review by henningandersen — COMMENTED** [review]
I would like to change the message format a bit, otherwise this looks good to me.

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/25_id_generation.yml [hunk 1]
```diff
@@ -186,6 +186,44 @@ index a new document on top of an old one over bulk:
...
   - match: {aggregations.max_rx.value: 4.30605511E8}
 
+---
+create operation on top of old document fails:
+  - skip:
+      version: " - 8.1.99"
+      reason: id generation changed in 8.2
+
+  - do:
+      catch: "/\\[.+/\\{.+\\}\\@2021-04-28T18:51:03.142Z\\]: version conflict, document already exists \\(current version \\[1\\]\\)/"
```
[the remark this answers is not in the document]
Comment (nik9000) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/25_id_generation.yml:196]: Sure!

**Comment by nik9000** [comment]
run elasticsearch-ci/part-1

**Review by romseygeek — APPROVED** [review]
LGTM

**Review by csoulios — APPROVED** [review]
LGTM!

**Comment by nik9000** [comment]
Thanks folks!

**Pull request merged**